src/arrow1/reader_async.rs

```diff
@@ -0,0 +1,33 @@
+use crate::arrow1::error::Result;
+use crate::common::fetch::{create_reader, get_content_length};
+use crate::log;
+
+use arrow::ipc::writer::StreamWriter;
+use arrow::record_batch::RecordBatchReader;
+use futures::stream::StreamExt;
+use parquet::arrow::async_reader::ParquetRecordBatchStreamBuilder;
+
+use async_compat::CompatExt;
+
+pub async fn read_parquet(url: String) -> Result<Vec<u8>> {
```

Comment:
Nice! I've been wanting to have an async api for arrow1 for a while. Especially now that arrow2 is slowing development